distributed/worker.py

```diff
@@ -811,6 +816,12 @@ def __init__(
             else dask.config.get("distributed.worker.memory.pause")
         )
 
+        self.max_spill = (
+            parse_bytes(max_spill)
+            if max_spill is not None
+            else dask.config.get("distributed.worker.memory.max-spill")
+        )
```

Comment:
Doesn't work with False.
```suggestion
        if max_spill is None:
            max_spill = dask.config.get("distributed.worker.memory.max-spill")
        self.max_spill = False if max_spill is False else parse_bytes(max_spill)
```